04-spatial-operations.Rmd

```diff
@@ -2,15 +2,16 @@
 
 ## Prerequisites {-}
 
-- This chapter requires the packages **tidyverse**, **sf** **raster**, and **spData**:
+- This chapter requires the packages **tidyverse**, **sf** and **raster**. 
+To avoid confusing function names (e.g., `dplyr::select` and `raster::select`) we will wait with attaching the **raster** package until we reach section \@ref{spatial-operations-on-raster-data):
```

Comment:
Reword to fix English grammar. I suggest:

```
To avoid confusing function names (e.g., `dplyr::select` and `raster::select`) we will wait until section \@ref{spatial-operations-on-raster-data) between loading **raster**:
```

Also note: we have not explained what `::` is yet - in my comment I explain it before referring to functions in this way. Maybe a good candidate for a note? Also I think it's good to *always* use the `dplyr::select()` because it will likely (and frequently does for me) cause problems later down the line. So in summary I'm not in favour of this change.

Reply:
You have already written a note on this in c3. Can change this back but then we will probably also have to always use `dplyr::filter()`

Reply:
I've never had name clashes with filter - don't think it's a raster function...

Reply:
Agree it's kinda inconsistent to only do it with select though. Will ask @Nowosad who's right here!

Reply:
There isn't a right answer here in my opinion. We should decide on a one way of doing that though.

Reply:
Another note: without `dplyr::` it was failing locally, even when I loaded **raster** later. So my preference is to keep it that way - minimises confusion all round I think.